splunk_handler/__init__.py

```diff
@@ -38,27 +39,33 @@ def emit(self, record):
 
         thread.start()
 
-    def _init_client(self):
-
-        return client.connect(
-                   host=self.host,
-                   port=self.port,
-                   username=self.username,
-                   password=self.password)
-
     def _async_emit(self, record):
 
-        global _client
-
-        if not _client:
-            _client = self._init_client()
-
         try:
-            _client.indexes[self.index].submit(
-                 self.format(record),
-                 host=socket.gethostname(),
-                 source=record.pathname,
-                 sourcetype='json')
+
+            if self.source is None:
+                source = record.pathname
+            else:
+                source = self.source
+
+            params = {
+                'host': self.hostname,
+                'index': self.index,
+                'source': source,
+                'sourcetype': self.sourcetype
+            }
+            url = 'https://%s:%s/services/receivers/simple' % (self.host, self.port)
+            payload = self.format(record)
```

Comment:
So should the a multiline python string look like this when escaped before being sent to Splunk?
![screen shot 2015-01-15 at 8 36 45 pm](https://cloud.githubusercontent.com/assets/2359186/5770790/641f2132-9cf6-11e4-8f75-3860dcedf4b0.png)

With this as the source:
```
message = """line one
             line two
             line three"""
```